Source/FolioReaderPage.swift

```diff
@@ -75,200 +97,208 @@ open class FolioReaderPage: UICollectionViewCell, UIWebViewDelegate, UIGestureRe
+                var locator = item.contentPre + item.content
                 locator += item.contentPost
-				locator = Highlight.removeSentenceSpam(locator) /// Fix for Highlights
-				let range: NSRange = tempHtmlContent.range(of: locator, options: .literal)
-
-				if range.location != NSNotFound {
-					let newRange = NSRange(location: range.location + item.contentPre.characters.count, length: item.content.characters.count)
-					tempHtmlContent = tempHtmlContent.replacingCharacters(in: newRange, with: tag) as NSString
-				}
-				else {
-					print("highlight range not found")
-				}
-			}
-		}
-		return tempHtmlContent as String
-	}
+                locator = Highlight.removeSentenceSpam(locator) /// Fix for Highlights
+                let range: NSRange = tempHtmlContent.range(of: locator, options: .literal)
+
+                if range.location != NSNotFound {
+                    let newRange = NSRange(location: range.location + item.contentPre.characters.count, length: item.content.characters.count)
+                    tempHtmlContent = tempHtmlContent.replacingCharacters(in: newRange, with: tag) as NSString
+                } else {
+                    print("highlight range not found")
+                }
+            }
+        }
+        return tempHtmlContent as String
+    }
 
     // MARK: - UIWebView Delegate
-    
+
     open func webViewDidFinishLoad(_ webView: UIWebView) {
-		guard let webView = webView as? FolioReaderWebView else {
-			return
-		}
+        guard let webView = webView as? FolioReaderWebView else {
+            return
+        }
 
-		delegate?.pageWillLoad?(self)
+        delegate?.pageWillLoad?(self)
 
-		// Add the custom class based onClick listener
-		self.setupClassBasedOnClickListeners()
+        // Add the custom class based onClick listener
+        self.setupClassBasedOnClickListeners()
 
         refreshPageMode()
-        
-        if readerConfig.enableTTS && !book.hasAudio() {
+
+        if (self.readerConfig.enableTTS == true && self.book.hasAudio() == false) {
             webView.js("wrappingSentencesWithinPTags()")
-            
-            if let audioPlayer = FolioReader.shared.readerAudioPlayer , audioPlayer.isPlaying() {
+
+            if let audioPlayer = self.folioReader.readerAudioPlayer, (audioPlayer.isPlaying() == true) {
                 audioPlayer.readCurrentSentence()
             }
         }
-        
-        let direction: ScrollDirection = FolioReader.needsRTLChange ? .positive() : .negative()
-        
-        if pageScrollDirection == direction && readerConfig.scrollDirection != .horizontalWithVerticalContent {
+
+        let direction: ScrollDirection = self.folioReader.needsRTLChange ? .positive(withConfiguration: self.readerConfig) : .negative(withConfiguration: self.readerConfig)
+
+        if (self.folioReader.readerCenter?.pageScrollDirection == direction && self.readerConfig.scrollDirection != .horizontalWithVerticalContent) {
```

Comment:
There is a bug here, when you go to table of contents and select a new chapter it should not scroll down as it doing now